src/ocm/components/clusterConfiguration/staticIp/data/formDataToInfraEnvField.ts

```diff
@@ -74,36 +76,44 @@ const getNmstateObject = (
         ),
       };
       nicName = getDummyNicName(protocolVersion);
-      interfaces.push(getEthernetInterface(nicName, protocolConfigs));
+
+      if (networkWide.useVlan && networkWide.vlanId) {
```

Comment:
I suggest a function
getInterface, that gets the name, protocolConfigs and networkWide.
It will return EthernetInterface | VlanInterface (actually these types can be combined now to type Interface with optional vlan section)
It should be reused in line 95
This will make this function much more readable, it's becoming hard to maintain